src/main/java/com/artipie/api/ApiChangeUserPassword.java

```diff
@@ -87,50 +79,25 @@ public Response response(final String line,
         return new AsyncResponse(
-            Single.fromCallable(() -> this.settings.meta().yamlMapping("credentials"))
-                .map(cred -> new KeyFromPath(cred.string("path")))
-                .flatMapCompletable(
-                    key ->
-                        Single.zip(
-                            new RxStorageWrapper(this.settings.storage())
-                                .value(key).to(ContentAs.YAML),
-                            Single.just(body).to(ContentAs.STRING).map(
-                                encoded -> URLEncodedUtils.parse(encoded, StandardCharsets.UTF_8)
-                                    .stream()
-                                    .filter(pair -> pair.getName().equals("password"))
-                                    .map(NameValuePair::getValue)
-                                    .findFirst().orElseThrow()
-                            ),
-                            (YamlMapping yaml, String pass) -> {
-                                YamlMappingBuilder result = Yaml.createYamlMappingBuilder();
-                                final YamlMapping credentials = yaml.yamlMapping("credentials");
-                                final List<YamlNode> keep = credentials.keys().stream()
-                                    .filter(node -> !node.asScalar().value().equals(user)).collect(Collectors.toList());
-                                for (final YamlNode node : keep) {
-                                    result = result.add(node, credentials.value(node));
-                                }
-                                result = result.add(
-                                    user,
-                                    Yaml.createYamlMappingBuilder()
-                                        .add("pass", String.format("sha256:%s", DigestUtils.sha256Hex(pass)))
-                                        .build()
-                                );
-                                return Yaml.createYamlMappingBuilder()
-                                    .add("credentials", result.build()).build()
-                                    .toString();
-                            }
-                        ).flatMapCompletable(
-                            yaml -> new RxStorageWrapper(this.settings.storage())
-                                .save(key, new Content.From(yaml.getBytes(StandardCharsets.UTF_8))))
+            Single.just(body).to(ContentAs.STRING).map(
+                encoded -> URLEncodedUtils.parse(encoded, StandardCharsets.UTF_8)
+                    .stream()
+                    .filter(pair -> pair.getName().equals("password"))
+                    .map(NameValuePair::getValue)
+                    .findFirst().orElseThrow()
+            ).flatMapCompletable(
+                pass -> Completable.fromFuture(
+                    this.settings.credentials().thenCompose(
+                        cred -> cred.add(user, pass, Credentials.PasswordFormat.SHA256)
```

Comment:
@olenagerasimova it seems that `pass` is stored in plain no matter what `PasswordFormat` is. It looks like a bug

Reply:
@olegmoz yes, you are right